python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,289 @@
+                    f"within tolerance in {chill_time} s"
+                )
+                break
+            else:
+                continue
+        else:
+            raise TimeoutError(
+                f"Gave up waiting after {self.timeout_chiller_cool_down}"
+                f"for the chiller to chill to {self.chiller_temperature}."
+                f"Stayed at {tel_chiller_temp}"
+            )
+
+    async def open_white_light_shutter(self):
+        await self.white_light_source.cmd_openShutter.set_start()
+        # Q? Should we include a check that the shutter is open?
+
+    async def switch_lamp_on(self):
+        await self.white_light_source.cmd_turnLampOn.set_start(
+            power=self.whitelight_power, timeout=self.timeout_lamp_warm_up
+        )
+
+    async def wait_for_lamp_to_warm_up(self):
+        self.white_light_source.evt_lampState.flush()
+        start_time_lamp_warm_up = time.time()
+        while time.time() - start_time_lamp_warm_up < self.timeout_lamp_warm_up:
+            lamp_state = await self.white_light_source.evt_lampState.next(
+                flush=False, timeout=self.timeout_lamp_warm_up
+            )
+            if lamp_state.basicState == ATWhiteLight.LampBasicState.WARMUP:
+                warmup_time_left = (
+                    lamp_state.warmupEndTime - lamp_state.private_sndStamp
+                )
+                self.log.info(f"Time Left for lamp warmup: {warmup_time_left} min.")
+                await asyncio.sleep(self.track_lamp_warmup)
+
+            elif lamp_state.basicState == ATWhiteLight.LampBasicState.ON:
+                self.log.info(
+                    f"White Light Lamp is on after a warm up of {time.time() - start_time_lamp_warm_up} s"
+                )
+        else:
+            raise TimeoutError(
+                f"White Light Lamp failed to turn on after {self.timeout_lamp_warm_up} s"
+            )
+
+    async def configure_atmonochromator(self):
+        await self.monochromator.cmd_selectGrating.set_start(
+            gratingType=self.grating_type, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeWavelength.set_start(
+            wavelength=self.wavelength, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeSlitWidth.set_start(
+            slit=1, slitWidth=self.entrance_slit_width, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeSlitWidth.set_start(
+            slit=2, slitWidth=self.exit_slit_width, timeout=self.cmd_timeout
+        )
+
+        params = await self.get_monochromator_parameters()
+
+        self.log.info(
+            f"ATMonochromator grating is {params[0]}, "
+            f"wavelength is {params[1]} nm "
+            f"with entry slit width {params[2]} "
+            f"and exit slit width {params[3]}"
+        )
+
+    async def get_monochromator_parameters(self):
+        tmp1 = await self.monochromator.evt_selectedGrating.aget()
```

Comment:
Please, add a timeout for all these `aget` statements, otherwise it can block here for ever.

Also you can probably do this in "in parallel" using `asyncio.gather`:

```
return await asyncio.gather(
    self.monochromator.evt_selectedGrating.aget(timeout=self.std_timeout),
    self.monochromator.evt_wavelength.aget(timeout=self.std_timeout),
    self.monochromator.evt_entrySlitWidth.aget(timeout=self.std_timeout),
    self.monochromator.evt_exitSlitWidth.aget(timeout=self.std_timeout),
)
```